superset/viz.py

```diff
@@ -1405,21 +1405,55 @@ def query_obj(self) -> QueryObjectDict:
         return {}
 
     def get_data(self, df: pd.DataFrame) -> VizData:
-        fd = self.form_data
-        # Late imports to avoid circular import issues
-        from superset import db
-        from superset.models.slice import Slice
-
-        slice_ids1 = fd.get("line_charts")
-        slices1 = db.session.query(Slice).filter(Slice.id.in_(slice_ids1)).all()
-        slice_ids2 = fd.get("line_charts_2")
-        slices2 = db.session.query(Slice).filter(Slice.id.in_(slice_ids2)).all()
-        return {
-            "slices": {
-                "axis1": [slc.data for slc in slices1],
-                "axis2": [slc.data for slc in slices2],
-            }
-        }
+        multiline_fd = self.form_data
+        # Late import to avoid circular import issues
+        from superset.charts.dao import ChartDAO
+
+        axis1_charts = ChartDAO.find_by_ids(multiline_fd.get("line_charts", []))
+        axis2_charts = ChartDAO.find_by_ids(multiline_fd.get("line_charts_2", []))
```

Comment:
It might be possible to combine these two queries into one:

```python
axis1_chart_ids = multiline_fd.get("line_charts", [])
axis2_chart_ids = multiline_fd.get("line_charts_2", [])
all_charts = ChartDAO.find_by_ids(axis1_chart_ids + axis2_chart_ids)
axis1_charts = all_charts[:len(axis1_chart_ids)]
axis2_charts = all_charts[len(axis1_chart_ids):]
```

Reply:
I did a slight variation of this to that doesn't fail if there are duplicates (shouldn't be the case, but you never know).